test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,203 @@
+	}
+	defer db.Close()
+	row, err := db.Query("SELECT * FROM device WHERE id=1234567890", 0)
+	if err != nil {
+		common.Failf("Query Sqlite DB failed: %v", err)
+	}
+	defer row.Close()
+	for row.Next() {
+		err = row.Scan(&device.id, &device.name, &device.description, &device.state, &device.lastOnline)
+		if err != nil {
+			common.Failf("Failed to scan DB rows: %v", err)
+		}
+		if string(device.id) == deviceId {
+			break
+		}
+	}
+	return device.state
+}
+
+//Run Test cases
+var _ = Describe("Event Bus Testing", func() {
+	var Token_client Token
+	var ClientOpts *MQTT.ClientOptions
+	var Client MQTT.Client
+	Context("Publish on eventbus topics throgh MQTT internal broker", func() {
+		BeforeEach(func() {
+			ClientOpts = HubClientInit(ctx.Cfg.MqttEndpoint, ClientID, "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if Token_client = Client.Connect(); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Connect() Error is %s", Token_client.Error())
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+		})
+		AfterEach(func() {
+			common.PrintTestcaseNameandStatus()
+		})
+
+		It("TC_TEST_EBUS_1: Sending data to Cloud", func() {
+			if Token_client = Client.Publish(UploadRecordToCloud, 0, false, "messagetoUpload_record_to_cloud"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_2: Sending data to device module", func() {
+			if Token_client = Client.Publish(DevicestatusUpdate, 0, false, "messagetoDevice_status_update"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_3: Sending data to device twin module", func() {
+			if Token_client = Client.Publish(DeviceTwinUpdate, 0, false, "messagetoDevice_Twin_update"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_4: Sending data to membership module", func() {
+			if Token_client = Client.Publish(DeviceMembershipUpdate, 0, false, "messagetoDevice_Membership_update"); Token_client.Wait() && Token_client.Error() != nil {
+				common.Failf("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_5: Sending data to device event module", func() {
```

Comment:
device event module ? which is this module ?

Reply:
@rohitsardesai83 , its a device module, modified the description.